app/javascript/src/Users/utils/index.jsx

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import type { Feature } from 'Users/types'
+
+const FEATURE_NAMES: { [feature: Feature]: string } = {
+  portal: 'Developer Portal',
+  finance: 'Billing',
+  settings: 'Settings',
+  partners: 'Developer Accounts -- Applications',
+  monitoring: 'Analytics',
+  plans: 'Integration & Application Plans',
+  policy_registry: 'Policy Registry'
+}
+
+const FEATURES_GRANTING_SERVICE_ACCESS = ['partners', 'monitoring', 'plans']
+
+export function getFeatureName (feature: Feature): string {
+  if (feature in FEATURE_NAMES) {
+    return FEATURE_NAMES[feature]
+  }
+
+  throw new Error(`${feature} is not a known feature`)
+}
+
+export function isServicePermissionsGranted (features: Feature | 'services' | Array<Feature | 'services'>): boolean {
+  if (typeof features === 'string') {
+    return FEATURES_GRANTING_SERVICE_ACCESS.includes(features)
+  } else {
+    return !!features.find(isServicePermissionsGranted)
+  }
+}
+
+export function toggleElementInCollection<T> (el: T, collection: T[]): T[] {
```

Comment:
What about ...
```
const i = collection.indexOf(el)
return (i > -1) ? [...collection, el] : collection.filter((_, j) => j != i)
```
?

Reply:
It's less readable IMO, also it takes twice as much time in Firefox, about the same time in Chrome. I like it but it uses complex syntax for such a small method.

Reply:
IMHO, it's more elegant and the benchmark you used it's not reliable, since it would depend on the addons and settings of your browsers (and possibly the resources in use in your machine). Also, it's just in the order of nanoseconds, it's not used in a non tail recursive operation or any in that case and it's just 2 lines of code against 10. But since this is not at all a dealbreaker to me, I'll leave it sink in your mind and/or maybe @damianpm or anyone else can give their opinion?

Reply:
I'd prefer the shorter syntax. It's true it's more cryptic, but you reduce to only 2 lines

Reply:
_Shorter + cryptic_ vs. _longer + descriptive_, I think we all know who's the winner. But since it's already 2 against me I won't hold this any longer.

Reply:
suck it up, and carry on...